boot/freeldr/freeldr/ntldr/winldr.c

```diff
@@ -35,6 +35,7 @@ BOOLEAN SafeBoot = FALSE;
 BOOLEAN BootLogo = FALSE;
 BOOLEAN NoexecuteDisabled = FALSE;
 BOOLEAN NoexecuteEnabled = FALSE;
+BOOLEAN PaeModeOn = FALSE;
```

Comment:
PaeEnabled - TRUE if BootOptions include "PAE"
PaeDisabled - TRUE if BootOptions include "NOPAE"
PaeModeOn - TRUE if all tests and checks alow enable PAE mode PaeModeOn ("traffic light green")

Reply:
My suggestion to fix that: Since the new `WinLdrIsPaeSupported` function is called just after parsing some boot options, modify that `WinLdrIsPaeSupported` function to pass to it the boot load options string, and move the parsing of the PAE switches inside. This will allow this function to calculate *THE* actual resulting PAE Mode being ON or OFF, depending on:
- the presence or absence of "PAE" switch;
- the presence or absence of "NOPAE" switch;
- and the rest of the checks you have mentioned (CPU support, etc, and kernel/HAL files).

Thus, the only remaining global BOOLEAN for Pae would be (say) `PaeModeOn`, while the two others would stay local temporary variables inside that function, holding the result of the boot options parsing.

Reply:
What about:
BOOLEAN NoexecuteDisabled = FALSE;
BOOLEAN NoexecuteEnabled = FALSE;

Reply:
These two Noexecute variables are redundant as well.